app/components/avo/views/resource_edit_component.rb

```diff
@@ -18,15 +26,13 @@ def title
   end
 
   def back_path
+    return back_path_from_bulk_update if from_bulk_update?
+
     # The `return_to` param takes precedence over anything else.
     return params[:return_to] if params[:return_to].present?
-
-    return resource_view_path if via_resource?
-    return resources_path if via_index?
-
-    if is_edit? && Avo.configuration.resource_default_view.show? # via resource show or edit page
-      return helpers.resource_path(record: @resource.record, resource: @resource, **keep_referrer_params)
-    end
+    return back_path_from_via_resource if via_resource?
+    return back_path_from_via_index if via_index?
+    return back_path_from_edit if is_edit? && Avo.configuration.resource_default_view.show?
```

Comment:
Why are we doing this refactor?

If this refactor was introduced due to CodeClimate warnings, let's ignore them for now and revert the change.

For feature PRs, it’s important to keep changes strictly relevant to the feature itself.

If refactor opportunities are identified during development, they should be proposed in a separate issue and addressed in a dedicated PR.

This practice helps maintain clear version history, simplifies tracking changes, and makes reverting specific work much easier if needed.